### PR TITLE
Bump jenkins core baseline

### PR DESCRIPTION
**Pull request by timja**
Initial bump needed for configuration as code support, tests are failing locally, seeing if any different on CI

Incompatibility with JCasC so far is the credential in this has the same symbol as the azure credentials plugin

### CODE REVIEW

**Comment by timja** [comment]
@gavinfish could you take a look please?

File: src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/AzureCredentialsTest.java [hunk 1]
```diff
@@ -86,15 +84,6 @@ public void testGetStorageAccountName() {
         assertEquals(stName, azureCred.getStorageAccountName());
     }
 
-    /**
-     * Test of getStorageKey method, of class AzureCredentials.
-     */
-    @Test
-    public void testGetStorageKey() {
```
[the remark this answers is not in the document]
Comment (timja) [reply src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/AzureCredentialsTest.java]: encrypted value was not the same as expected,
the method of generation was changed in the years between jenkins versions.

comparing the encrypted version is not that interesting, what is interesting is that the plain text version is correct

File: src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/CredentialMigrationTest.java [hunk 2]
```diff
@@ -98,7 +98,7 @@ public void testUpgradeStorageConfig() throws Exception {
 
         assertEquals(u.getStorageAccountName(), storageCred.getStorageAccountName());
         assertEquals(u.getEndpointURL(), storageCred.getBlobEndpointURL());
-        assertEquals(u.getSecureKey().getEncryptedValue(), storageCred.getStorageKey());
```
[the remark this answers is not in the document]
Comment (timja) [reply src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/CredentialMigrationTest.java]: encrypted value was not the same as expected,
the method of generation was changed in the years between jenkins versions.

comparing the encrypted version is not that interesting, what is interesting is that the plain text version is correct

File: src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/AzureCredentialsTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (gavinfish) [reply src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/AzureCredentialsTest.java]: I have checked the code of the jenkins core. This [change](https://github.com/jenkinsci/jenkins/blob/850168fdb2ef99c5d6fef0b2819cb490ed690fe8/core/src/main/java/hudson/util/Secret.java#L130-L134) will cause using a random string to salt the secret if the variable `iv` has not been set.

[the remark this answers is not in the document]
Comment (gavinfish) [reply src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/AzureCredentialsTest.java]: How about taking  this test case back with comparison with plain text?

[the remark this answers is not in the document]
Comment (timja) [reply src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/AzureCredentialsTest.java]: done

**Comment by timja** [comment]
@gavinfish green build now
The tests were failing on windows before due to JEP-200 issues 
https://jenkins.io/blog/2018/01/13/jep-200/

most likely due to the weird inner class for the credentials.
it's a difficult fix in a compatible way and should be done separately to this change I think.

**Pull request merged**